.travis.yml

```diff
@@ -0,0 +1,53 @@
+dist: trusty
+sudo: false
+language: php
+
+cache:
+  directories:
+    - $HOME/.composer/cache
+
+install:
+  - composer require typo3/cms="$TYPO3_VERSION"
+
+jobs:
+  include:
+    - php: 7.1
+      env: TYPO3_VERSION=^8.7
+    - php: 7.2
+      env: TYPO3_VERSION=^8.7
+    - php: 7.2
+      env: TYPO3_VERSION=^9.0
+
+    - &lint
+      php: 7.1
+      env: Other lints
+      install:
+        - composer install
+      script:
+        - composer lint:typoscript
+        - composer lint:xml
+    - <<: *lint
+      env: Codestyle
+      script:
+        - composer lint:style
+
+    - stage: deploy
+      if: tag IS present
+      php: 7.1
+      before_install: skip
+      install: skip
+      before_script: skip
+      script: |
+        echo -e "Preparing upload of release ${TRAVIS_TAG} to TER\n"
+
+        TAG_ANNOTATION="$(git tag -n -l $TRAVIS_TAG)"
+        TAG_MESSAGE="${TAG_ANNOTATION#* }"
+
+        git reset --hard
+        git clean -xfd
+
+        export PATH=$PATH:$(composer global config bin-dir --absolute 2>/dev/null)
+        composer global require helhum/ter-client dev-master
+
+        echo "Uploading release ${TRAVIS_TAG} to TER"
+        ter-client upload $(composer config extra.typo3/cms.extension-key) . -u "$TYPO3_ORG_USERNAME" -p "$TYPO3_ORG_PASSWORD" -m "$TAG_MESSAGE"
```

Comment:
About the TER upload, do I still need to define an extension key ? Currently I have the mjml one

Reply:
This script will fetch the key from the `composer.json`, you have to define it in the `extra` section, sub section `typo3/cms` as `extension-key`.
Or you can hardcode it in the script.